driver/src/main/java/org/neo4j/driver/DriverQueryConfig.java

```diff
@@ -0,0 +1,40 @@
+package org.neo4j.driver;
+
+import org.neo4j.driver.internal.BookmarksHolder;
+
+import java.util.Set;
+
+public record DriverQueryConfig(String database, Set<Bookmark> bookmarks, SessionQueryConfig sessionQueryConfig) {
+
+    public static DriverQueryConfigBuilder builder() {
+        return new DriverQueryConfigBuilder();
+    }
+
+    public static final DriverQueryConfig defaultInstance =
+            new DriverQueryConfig("neo4j", null, SessionQueryConfig.defaultInstance);
+
+    public static final DriverQueryConfig read = new DriverQueryConfig("neo4j", null, SessionQueryConfig.read);
+
+    public static final DriverQueryConfig write = new DriverQueryConfig("neo4j", null, SessionQueryConfig.write);
+
+    public static final DriverQueryConfig autoCommit =
+            new DriverQueryConfig("neo4j", null, SessionQueryConfig.autoCommit);
+
+    public boolean validate() {
+        return sessionQueryConfig.validate();
+    }
+
+    public SessionConfig sessionConfig(BookmarksHolder bookmarksHolder) {
```

Comment:
If this is for public API, then `BookmarksHolder` should not be used here since it is an internal class.